### PR TITLE
DRY up prequisites and leverage them to create deployments; Add code for NGINX confs

### PR DESCRIPTION
**Pull request by puneetsarna**

### CODE REVIEW

File: snippets/terraform/configurations/main.tf [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  tags = var.tags
+}
+
+resource "azurerm_nginx_configuration" "example" {
+  nginx_deployment_id = azurerm_nginx_deployment.example.id
+  root_file           = "/etc/nginx/nginx.conf"
+
+  config_file {
+    content = base64encode(<<-EOT
```
Comment (agazeley) [inline snippets/terraform/configurations/main.tf:50]: Could we make this read in an actual file somehow? That is going to be closer to what a customer actually will do.

File: snippets/terraform/deployments/create-or-update/main.tf [hunk 2]
```diff
@@ -12,98 +12,31 @@ provider "azurerm" {
...
   name                     = var.name
-  resource_group_name      = azurerm_resource_group.example.name
+  resource_group_name      = module.prerequisites.resource_group_name
   sku                      = var.sku
-  location                 = azurerm_resource_group.example.location
+  location                 = var.location
   managed_resource_group   = "example"
   diagnose_support_enabled = true
 
   identity {
```
Comment (agazeley) [inline snippets/terraform/deployments/create-or-update/main.tf:30]: We are not giving the MI any role assignments? Is this intentional?

Comment (puneetsarna) [reply snippets/terraform/deployments/create-or-update/main.tf:30]: 🤦‍♂️ completely forgot that it's not auto-done like the portal. thanks for catching it.

Comment (agazeley) [reply snippets/terraform/deployments/create-or-update/main.tf:30]: I really wish it was done that way all the time.

Comment (puneetsarna) [reply snippets/terraform/deployments/create-or-update/main.tf:30]: done.

File: snippets/terraform/configurations/main.tf (hunk 1 above)
Comment (puneetsarna) [reply snippets/terraform/configurations/main.tf]: yep. done.

**Review by agazeley — APPROVED** [review]

**Pull request merged**